### DISCUSSION TITLE
How does this app theme GDM?

### DISCUSSION THREAD

**Discussion by schuelermine** [realmazharhussain/gdm-settings#115] [Q&A] [answered]
Hi, I’ve searched around the internet for a way to theme modern GDM and I’ve only found this app. All other resources are outdated. Now I’m interested in how exactly this app does this. Because my system is configured in an unusual manner, I can’t actually use this app to theme my GDM. How does the app change the GDM shell & icon theme? Is it a special file? Or a dconf entry for root? Or something else? I’d be delighted to hear more.

**Comment by realmazharhussain** [comment, marked as answer]
> Is it a special file?

Yes, to change shell theme, we need to modify `/usr/share/gnome-shell/gnome-shell-theme.gresource` file.

> Or a dconf entry for root?

Yes, to change most of the settings (like icon theme), we need to add/modify files in `/etc/dconf/db/gdm.d/` and run `sudo dconf update`.

Full details can be found at <https://github.com/realmazharhussain/gdm-settings/discussions/87#discussioncomment-3378018>.

**Reply by schuelermine** [reply]
Thank you!

**Comment by realmazharhussain** [comment]
> my system is configured in an unusual manner, I can’t actually use this app to theme my GDM.

I would like to know what is unusual about your system configuration. Maybe, I could add support for it in my app?

**Reply by schuelermine** [reply]
I'm using NixOS so my system is configured declaratively via a Nix language expression. I have no /usr and all /etc files are symlinks to a read-only mounted store where every system component is identified by a hash.

**Reply by schuelermine** [reply]
Actually, if your app had a command-line version that would be useful as it could then be used to automatically build a modified gnome shell file. I will use the bash script you provided in your other answer for now (Nix & nixpkgs are basically a very fancy way to glue bash scripts together, but sandboxed)

**Reply by realmazharhussain** [reply]
> Actually, if your app had a command-line version that would be useful as it could then be used to automatically build a modified gnome shell file.

Before knowing how to write GUI applications, I wrote a bash version of this app. It is available at <https://github.com/realmazharhussain/gdm-tools>. But I doubt that it'll be useful since it works the same way as this app behind the scenes and file locations are hard-coded to be in `/etc` and `/usr`.
